testprograms/test_ppc_const_int128.c

```diff
@@ -0,0 +1,135 @@
+
+
+const vui128_t vtipowof10[] = { (vui128_t) (__int128 ) 1ll, /* 10**0 */
+  (vui128_t) (__int128 ) 10ll, /* 10**1 */
+  (vui128_t) (__int128 ) 100ll, /* 10**2 */
+  (vui128_t) (__int128 ) 1000ll, /* 10**3 */
+  (vui128_t) (__int128 ) 10000ll, /* 10**4 */
+  (vui128_t) (__int128 ) 100000ll, /* 10**5 */
+  (vui128_t) (__int128 ) 1000000ll, /* 10**6 */
+  (vui128_t) (__int128 ) 10000000ll, /* 10**7 */
+  (vui128_t) (__int128 ) 100000000ll, /* 10**8 */
+  (vui128_t) (__int128 ) 1000000000ll, /* 10**9 */
+  CONST_VUINT128_QxW (0, 0, 0x2, 0x540be400),  /* 10**10 */
+  CONST_VUINT128_QxW (0, 0, 0x17, 0x4876e800),  /* 10**11 */
+  CONST_VUINT128_QxW (0, 0, 0xe8, 0xd4a51000),  /* 10**12 */
+  CONST_VUINT128_QxW (0, 0, 0x918, 0x4e72a000),  /* 10**13 */
+  CONST_VUINT128_QxW (0, 0, 0x5af3, 0x107a4000),  /* 10**14 */
+  CONST_VUINT128_QxW (0, 0, 0x38d7e, 0xa4c68000),  /* 10**15 */
+  CONST_VUINT128_QxW (0, 0, 0x2386f2, 0x6fc10000),  /* 10**16 */
+  CONST_VUINT128_QxW (0, 0, 0x1634578, 0x5d8a0000),  /* 10**17 */
+  CONST_VUINT128_QxW (0, 0, 0xde086b3, 0xa7640000),  /* 10**18 */
+  (vui128_t) (__int128 ) 10ll * (__int128 ) 1000000000000000000ll, /* 10**19 */
+  (vui128_t) (__int128 ) 100ll * (__int128 ) 1000000000000000000ll, /* 10**20 */
+  (vui128_t) (__int128 ) 1000ll * (__int128 ) 1000000000000000000ll, /* 10**21 */
+  (vui128_t) (__int128 ) 10000ll * (__int128 ) 1000000000000000000ll, /* 10**22 */
+  (vui128_t) (__int128 ) 100000ll * (__int128 ) 1000000000000000000ll, /* 10**23 */
+  (vui128_t) (__int128 ) 1000000ll * (__int128 ) 1000000000000000000ll, /* 10**24 */
+  (vui128_t) (__int128 ) 10000000ll * (__int128 ) 1000000000000000000ll, /* 10**25 */
+  (vui128_t) (__int128 ) 100000000ll * (__int128 ) 1000000000000000000ll, /* 10**26 */
+  (vui128_t) (__int128 ) 1000000000ll * (__int128 ) 1000000000000000000ll, /* 10**27 */
+  (vui128_t) (__int128 ) 10000000000ll * (__int128 ) 1000000000000000000ll, /* 10**28 */
+  (vui128_t) (__int128 ) 100000000000ll * (__int128 ) 1000000000000000000ll, /* 10**29 */
+  (vui128_t) (__int128 ) 1000000000000ll * (__int128 ) 1000000000000000000ll, /* 10**30 */
+  (vui128_t) (__int128 ) 10000000000000ll * (__int128 ) 1000000000000000000ll, /* 10**31 */
+  CONST_VUINT128_Qx16d (10000000000000000UL, 0), /* 10**32 */
+  (vui128_t) (__int128 ) 1000000000000000ll * (__int128 ) 1000000000000000000ll, /* 10**33 */
+  (vui128_t) (__int128 ) 10000000000000000ll * (__int128 ) 1000000000000000000ll, /* 10**34 */
+  (vui128_t) (__int128 ) 100000000000000000ll * (__int128 ) 1000000000000000000ll, /* 10**35 */
+  CONST_VUINT128_Qx18d (1000000000000000000UL, 0), /* 10**36 */
+  (vui128_t) (__int128 ) 10ll * (__int128 ) 1000000000000000000ll
+           * (__int128 ) 1000000000000000000ll, /* 10**37 */
+  CONST_VUINT128_Qx19d (10000000000000000000UL, 0), /* 10**38 */
+  (vui128_t) (__int128 ) 0L };/* end marker */
+
+#include <altivec.h>
+int main(int argc, char* argv[])
+{
+#if defined(__VSX__) && defined(__ALTIVEC__)
```

Comment:
nit: seems like these should be reversed, since ALTIVEC is more generic.

Reply:
OK